Source/Library.cpp

```diff
@@ -23,18 +29,32 @@ void Library::on_testLaunch_clicked()
     dialog.setWindowTitle("Select Executable");
     dialog.setFileMode(QFileDialog::ExistingFile);
 
-
     if (dialog.exec()) {
         QStringList files = dialog.selectedFiles();
         QString file = files.at(0);
-
+#ifdef Q_WS_MACX
+        //Get the binary from the app bundle
+        QDir dir(file + "/Contents/MacOS");
+        QStringList fileList = dir.entryList();
+        file = dir.absoluteFilePath(fileList.at(2));//USUALLY this is the executable (after ., ..)
+#endif
         runProcess(file);
     }
 }
 
 void Library::runProcess(QString location) {
     // TODO: Implement some threading
-    QProcess process(this);
-    process.start("\"" + location + "\"");
-    process.waitForFinished(-1);
+    if (!processRunning) {
+        runningProcess->start(location);
+        runningProcess->waitForStarted();
+        processRunning = true;
+    } else {
+        QMessageBox messageBox(this);
```

Comment:
Also, you probably want to display the message box before the QFileDialog pops up to select.